src/index.js

```diff
@@ -1,14 +1,25 @@
 import 'preact/debug'
 import { h, render } from 'preact'
 import Banana from 'banana-i18n'
-import { setAppLanguage, getAppLanguage, loadAllLanguagesMessages } from 'utils'
+import {
+  setAppLanguage, getAppLanguage, setDeviceLanguage, getDeviceLanguage,
+  checkHasDeviceLanguageChanged, loadAllLanguagesMessages
+} from 'utils'
 import { App } from 'components'
 
 import '../style/style.less'
 
-const lang = getAppLanguage() || navigator.language.substr(0, 2)
-setAppLanguage(lang)
-const banana = new Banana(lang)
+let banana
+
+if (checkHasDeviceLanguageChanged()) {
+  setDeviceLanguage()
+  const lang = getDeviceLanguage()
+  setAppLanguage(lang)
+  banana = new Banana(lang)
+} else {
+  const lang = getAppLanguage()
```

Comment:
You can probably reorder things here to have 2 blocks
1) reset the app language if the device language has changed
2) unconditionally get app language and create the banana instance